core/arch/riscv/include/riscv.h

```diff
@@ -0,0 +1,283 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2022 NXP
+ */
+
+#ifndef RISCV_H
+#define RISCV_H
+
+#include <compiler.h>
+#include <encoding.h>
+#include <sys/cdefs.h>
+#include <stdint.h>
+#include <util.h>
+
+#define RISCV_XLEN_BITS		(__riscv_xlen)
+#define RISCV_XLEN_BYTES	(__riscv_xlen / 8)
+
+#define REGOFF(x)			((x) * RISCV_XLEN_BYTES)
+
+#if __riscv_xlen == 32
+#define STR       sw
+#define LDR       lw
+#else
+#define STR       sd
+#define LDR       ld
+#endif
+
+#ifndef __ASSEMBLER__
+
+#define get_field(reg, mask) (((reg) & \
+                 (unsigned long)(mask)) / ((mask) & ~((mask) << 1)))
+
+#define set_field(val, which, fieldval) (((val) & ~(which)) | ((fieldval) * ((which) & ~((which)-1))))
+
+static inline __noprof void mb(void)
+{
+	asm volatile ("fence" : : : "memory");
+}
+
+static inline __noprof unsigned long read_tp(void)
+{
+	unsigned long tp;
+
+	asm volatile("mv %0, tp" : "=&r"(tp));
+	return tp;
+}
+
+static inline __noprof void wfi(void)
+{
+	asm volatile ("wfi");
+}
+
+static inline __noprof void flush_tlb(void)
+{
+	asm volatile("sfence.vma zero, zero");
+}
+
+static inline __noprof void flush_tlb_entry(unsigned long va)
+{
+	asm volatile ("sfence.vma %0" : : "r" (va) : "memory");
+}
+
+// supervisor address translation and protection
```

Comment:
Use C-style comments and initial uppercase `/* Supervisor address translation and protection */`